ui/packages/tidb-dashboard-lib/src/components/MetricChart/index.tsx

```diff
@@ -271,53 +297,58 @@ export default function MetricChart({
       <div style={{ height }}>
         <Space direction="vertical">
           <ErrorBar errors={[error]} />
-          <Link to="/user_profile?blink=profile.prometheus">
-            {t('components.metricChart.changePromButton')}
-          </Link>
+          {promAddrConfigurable && (
+            <Link to="/user_profile?blink=profile.prometheus">
+              {t('components.metricChart.changePromButton')}
+            </Link>
+          )}
         </Space>
       </div>
     )
   } else {
     inner = (
-      <Chart size={{ height }} ref={chartRef}>
-        <Settings
-          {...DEFAULT_CHART_SETTINGS}
-          legendPosition={Position.Right}
-          legendSize={130}
-          pointerUpdateDebounce={0}
-          onPointerUpdate={(e) => ee.emit(e)}
-        />
-        <Axis
-          id="bottom"
-          position={Position.Bottom}
-          showOverlappingTicks
-          tickFormat={timeTickFormatter(range)}
-        />
-        <Axis
-          id="left"
-          position={Position.Left}
-          showOverlappingTicks
-          tickFormat={(v) =>
-            unit ? getValueFormat(unit)(v, 2) : Number(v).toFixed(0)
-          }
-          ticks={5}
-        />
-        {data?.values.map((qd) => renderQueryData(type, qd))}
-        {data && (
-          <LineSeries // An empty series to avoid "no data" notice
-            id="_placeholder"
-            xScaleType={ScaleType.Time}
-            yScaleType={ScaleType.Linear}
-            xAccessor={0}
-            yAccessors={[1]}
-            hideInLegend
-            data={[
-              [data.meta.queryOptions.start * 1000, null],
-              [data.meta.queryOptions.end * 1000, null]
-            ]}
+      <>
+        {hasEmptyQueryResult && <BadgeWithTooltip />}
```

Comment:
Can we use the badge inside the chart? Otherwise, the position of the badge will depend on the outside containers. Like:

![image](https://user-images.githubusercontent.com/11549583/182086234-7df7c6de-7b0d-4a5f-9d14-dfd3ad90c4ab.png)